fv/wireguard_test.go

```diff
@@ -484,6 +484,7 @@ var _ = infrastructure.DatastoreDescribe("_BPF-SAFE_ WireGuard-Supported", []api
 
 			cc.ResetExpectations()
 
+			policy = api.NewGlobalNetworkPolicy()
```

Comment:
I agree that it's good practice to use a new GNP object here.  But do you know exactly what the problem was as the code was before, when it was reusing the same object?

Reply:
Thanks for taking a look @neiljerram. I agree this _should_ not be needed.

With the current proposed changes, I am trying to repro the flake we see with respect to workload connectivity when `Apply` policy is applied. In my experiment, without these changes, I was able to repro the flake in 1 out of 4 runs. With these changes however, I haven't been able to repro the flake with 10 successful runs so far. I am not sure atm what's "fixing" the flake. I have tried `5 seconds sleep` before `CheckConnectivity()` but that didn't address the flake ruling out the timing issue.

Can you think of any check to do before `CheckConnectivity()` other than reading the policy for verifying that the policy is applied, which is already being done?

Reply:
So I think you're saying that this empirically reduces the occurrence of the flake, but you don't yet know why.  Is that right?

Have you tried dumping and comparing the detailed `*policy` that is applied, with and without the addition of line 489 ?

Reply:
As regards sleeping before `CheckConnectivity()`,

- I think we should always avoid hardcoded sleeping, and instead use something like Eventually or an explicit retry loop

- it shouldn't be needed, because `CheckConnectivity()` already does the right kind of retrying, I think.